src/Http/GenericRouter.php

```diff
@@ -172,4 +160,68 @@ private function createResponse(Response|View $input): Response
 
         return $input;
     }
+
+    private function matchStaticRoute(Request $request): ?MatchedRoute
+    {
+        $staticRoute = $this->getRoutes()[$request->method->value][$request->getPath()] ?? null;
+
+        if ($staticRoute === null) {
+            return null;
+        }
+
+        // TODO: why do we need to check this?
+        if ($staticRoute->isDynamic) {
```

Comment:
@N-Silbernagel not a blocker, but I suppose we can get rid of this check if we split dynamic and static routes in the config?

Reply:
Yes, that would work, what I did not like about that, is that RouteConfig would not have a route param but maybe "staticRoutes" and "dynamicRoutes", which might not be ideal for reuse of the RouteConfig class.
On the other side, RouteConfig is probably only useful for the Router.

Reply:
That's a good point. Let's actually keep it as it is for now, we can always improve on it later if needed.